File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {

		int digit;
		int power = 0;
		int check = numberToCheck;
		int armstrong = 0;
		power = Integer.toString(Math.abs(numberToCheck)).trim().length();
		while ((check != 0) && (armstrong < numberToCheck)) {
			digit = check % 10;
			check = (check - digit) / 10;
			armstrong = armstrong + (int) Math.pow((double) digit, (double) power);
		}
		return (numberToCheck == armstrong);
	}

}

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        IEnumerable<int> digits = (from char c in number.ToString().Trim()
                        where c >= '0' && c <= '9'
                        select (int)c - (int)'0');
        int num_digits = digits.Count();
        return (from int digit in digits
                select Math.Pow(digit, num_digits)).Sum() == number;
    }
}